### PR TITLE
Add description of how the directory structure targets Python versions.

### PR DESCRIPTION
**Pull request by emilyemorehouse**
This is fairly self-explanatory. Simply adds information on how the directory structure is used to target Python versions along with an extension of the existing best practices for which directory to put stubs in.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -172,9 +172,20 @@ contributors.
 
 There are separate directories for `stdlib` and `third_party` stubs.
 Within those, there are separate directories for different versions of
-Python the stubs target.  If the given library works on both Python 2
-and Python 3, prefer to put your stubs in an `2and3` directory, unless
-the types are so different that the stubs become unreadable that way.
+Python the stubs target.
+
+As a convention, the directory name indicates the lowest major version of Python
```
Comment (JelleZijlstra) [inline CONTRIBUTING.md]: Remove "As a convention", it's redundant.

Also, I don't think "lowest" is right; when you're running mypy in Python 3 mode, it won't use stubs from stdlib/2.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -172,9 +172,20 @@ contributors.
...
+that a stub targets, with the exception of the `2and3` directory which applies
+to both Python 2 and 3. For example, stubs in the `3` directory will be applied
+to all versions of Python 3, though stubs in the `3.6` directory will only be
+applied to versions 3.6 and above.
+
+It is preferred to use a single stub in the more generic directory that
+conditionally targets specific versions when needed, as opposed
+to maintaining multiple stub files within more specific directories. Similarly,
+if the given library works on both Python 2 and Python 3, prefer to put your
+stubs in an `2and3` directory, unless the types are so different that the stubs
```
Comment (JelleZijlstra) [inline CONTRIBUTING.md]: an -> a

Comment (emilyemorehouse) [inline CONTRIBUTING.md]: After re-reading, this actually makes more sense as 'the'.

File: CONTRIBUTING.md (hunk 1 above)
Comment (emilyemorehouse) [inline CONTRIBUTING.md]: Is "lowest major version" unclear? (Perhaps so, as it was not clear enough for you). 
I could add an 'e.g. Python 2 or Python 3' if that doesn't seem redundant.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -172,9 +172,23 @@ contributors.
...
-the types are so different that the stubs become unreadable that way.
+Python the stubs target.
+
+The directory name indicates the major version of Python that a stub targets
+and optionally the lowest minor version, with the exception of the `2and3`
+directory which applies to both Python 2 and 3.
+
+For example, stubs in the `3` directory will be applied to all versions of
+Python 3, though stubs in the `3.6` directory will only be applied to versions
+3.6 and above. However, stubs in the `2` directory will not be applied to
```
Comment (gvanrossum) [inline CONTRIBUTING.md]: (Maybe could add that if 3/foo.pyi and 3.6/foo.pyi both exist, the latter wins on Python 3.6 but the former is used on Python 3.2 -- 3.5; similarly if 3.5/foo.pyi and 3.6/foo.pyi both exist.)

Comment (emilyemorehouse) [inline CONTRIBUTING.md]: Is there support for patch versions (i.e. 3.6.1) or only minor versions?

Comment (JelleZijlstra) [inline CONTRIBUTING.md]: Only minor versions. For patch versions we always use the most recent patch version in stubs.

**Pull request merged**